pkg/amqp-lib/AmqpConsumer.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Enqueue\AmqpLib;
+
+use Interop\Queue\InvalidMessageException;
+use Interop\Queue\PsrConsumer;
+use Interop\Queue\PsrMessage;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage as LibAMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+
+class AmqpConsumer implements PsrConsumer
+{
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * @var AmqpQueue
+     */
+    private $queue;
+
+    /**
+     * @param AMQPChannel $channel
+     * @param AmqpQueue   $queue
+     */
+    public function __construct(AMQPChannel $channel, AmqpQueue $queue)
+    {
+        $this->channel = $channel;
+        $this->queue = $queue;
+    }
+
+    /**
+     * @return AmqpQueue
+     */
+    public function getQueue()
+    {
+        return $this->queue;
+    }
+
+    /**
+     * @param int $timeout
+     *
+     * @return AmqpMessage|null
+     */
+    public function receive($timeout = 0)
+    {
+        $end = microtime(true) + ($timeout / 1000);
+
+        while (0 === $timeout || microtime(true) < $end) {
```

Comment:
would be able to add a support of `basic_consume`. You can look how we did it in amqp ext transport